src/core/ext/xds/xds_api.cc

```diff
@@ -1560,10 +1560,13 @@ grpc_error_handle RetryPolicyParse(
       }
     }
   }
-  // TODO(donnadionne): when we add support for per_try_timeout, we will need to
-  // return a policy if per_try_timeout is set even if retry_on specified no
-  // supported policies.
-  if (retry_to_return.retry_on.Empty()) return GRPC_ERROR_NONE;
+  if (retry_to_return.retry_on.Empty()) {
+    // we dont have any supported retry_on status, but we will still retrun an
+    // empty policy so that the per route policy takes precedence.  Returning a
+    // policy here also helps in the future where per_try_timeout is supported.
+    *retry = retry_to_return;
+    return GRPC_ERROR_NONE;
+  }
```

Comment:
Why not remove this block entirely and still perform validation for other fields?

Reply:
I agree, I think it would make more sense to validate the rest of the policy.

Reply:
removed block as suggested